src/PlaylistController.js

```diff
@@ -0,0 +1,103 @@
+import assert from 'assert';
+import GenericController from './GenericController';
+import Playlist from './Structs/Playlist';
+
+export default class PlaylistController extends GenericController {
+  constructor(...args) {
+    super(...args);
+    if (window.APPCONTEXT && window.APPCONTEXT.Go && window.APPCONTEXT.Go.h && window.APPCONTEXT.Go.h.length) {
+      this._playlists = Object.assign({}, window.APPCONTEXT.Go.h[0].Hi);
+      this._watchPlaylistObject();
+    } else {
+      this._playlists = {};
+    }
+
+    this.addMethod('getAll', this.getAll.bind(this));
+    this.addMethod('play', this.play.bind(this));
+    this.addMethod('playWithTrack', this.playWithTrack.bind(this));
```

Comment:
Ohhh, I see. I think we did that in GMusic so namespaces could talk to each
other if necessary (e.g. hooks to playback)
On May 30, 2016 9:41 AM, "Samuel Attard" <notifications@github.com> wrote:

> In src/PlaylistController.js
> <https://github.com/gmusic-utils/gmusic-ui.js/pull/1#discussion_r65080420>
> :
>
> > +import GenericController from './GenericController';
> > +import Playlist from './Structs/Playlist';
> > +
> > +export default class PlaylistController extends GenericController {
> > +  constructor(...args) {
> > +    super(...args);
> > +    if (window.APPCONTEXT && window.APPCONTEXT.Go && window.APPCONTEXT.Go.h && window.APPCONTEXT.Go.h.length) {
> > +      this._playlists = Object.assign({}, window.APPCONTEXT.Go.h[0].Hi);
> > +      this._watchPlaylistObject();
> > +    } else {
> > +      this._playlists = {};
> > +    }
> > +
> > +    this.addMethod('getAll', this.getAll.bind(this));
> > +    this.addMethod('play', this.play.bind(this));
> > +    this.addMethod('playWithTrack', this.playWithTrack.bind(this));
>
> Yes, because I don't want GMusic to change the scope away from the class
> (that wouldn't make sense).
>
> Each class method should be scoped to the instance of the class even when
> called from somewhere else (It's a class thing).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gmusic-utils/gmusic-ui.js/pull/1/files/429ab1578711c300bd6190b6faedb1b274a6d209#r65080420>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA3FWFxb_0dBsBmZ5MYRQIWmD-nqa3cNks5qGvcsgaJpZM4IpiKO>
> .
>